### ISSUE TITLE
Replace Dependabot with Renovate

### ISSUE THREAD

**Issue by alisonailea** [Esri/calcite-design-system#5746]
### Pain Point
Dependabot does not support multi-branch dependency updates

### Reasons for Renovate over Dependabot
- Renovate is industry standard
- Renovate has better configuration options

### Steps to resolve:
1. Install Renovate for Calcite-Components - https://github.com/apps/renovate
	a. This will automatically open a PR but no other changes will effect the repo until that PR is approved and merged
2. Configure Renovate - https://docs.renovatebot.com/configuration-options/
3. Get approval from Calcite engineers
4. Deactivate Dependabot
5. Merge Renovate PR
6. Profit

### Desired Outcome

Replace Dependabot with Renovate

### Resource Links

- https://github.com/apps/renovate
- https://github.com/renovatebot/renovate
- https://docs.renovatebot.com/getting-started/installing-onboarding/
- https://docs.renovatebot.com/configuration-options/

### Recommended config setup

```json
{
  "labels": ["dependencies"],
  "schedule": [
    "after 10pm every weekday",
    "before 5am every weekday"
  ],
  "baseBranches": ["main", "next", "beta"],
  "commitMessagePrefix": "build(deps): ",
  "ignoreDeps": ["chalk", "cheerio", "typescript", "@types/jest", "jest", "jest-cli", "ts-jest", "puppeteer"],
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true
    },
    {
      "matchPackagePatterns": ["eslint"],
      "labels": ["linting"]
    },
    {
      "matchDepTypes": ["optionalDependencies"],
      "addLabels": ["optional"]
    }
  ]
}
```

**Comment by geospatialem** [comment]
This came up as a possible mitigation for https://github.com/Esri/calcite-components/issues/6729.

Reference https://github.com/Esri/calcite-components/issues/6729#issuecomment-1496250527 for the use case on our CI, specific to floating-ui and stencil dependencies. 

Until CI can be adopted, Calcite team members will need to individually edit the dependabot-generated PR's and update them to `"feat(floating-ui): "`/`"feat(stencil)"`.

**Comment by geospatialem** [comment]
Assigning to the June release to conduct research prior to implementation. Research will include Renovate's feasibility, and determine next steps for implementation, if applicable.

**Comment by benelan** [comment]
> Assigning to the June release to conduct research prior to implementation. Research will include Renovate's feasibility, and determine next steps for implementation, if applicable.

Note Ali created this issue after I mentioned to her that I had researched and planned on switching us to Renovate eventually. I should really do a better job of creating issues for things I put on my personal list lol.

But yeah, next steps are just configuring Renovate and doing a bit of testing on my end. I already started putting together the config, so I should be able to get a PR in during the July sprint.
